### PR TITLE
Overriding Assignmentes Analysis

### PR DESCRIPTION
**Pull request by galilasmb**
I created some auxiliary classes and modified others. I also produced many tests.

### CODE REVIEW

File: src/main/java/br/unb/cic/analysis/df/OverridingAssignmentFieldsRefAnalysis.java [hunk 1]
```diff
@@ -194,7 +194,7 @@ private KeyAndFlowHash getKeyAndElementsOfHashAndFlowList(String nextKey, Set<Ha
             }
         }
         //If auxValuesHasMap is equal to zero and nextKey does not contain a "$stack", then it is the starting field of the object
-        if (!nextKey.contains("$stack") && (isNextKey)){
+        if (!(actualField==null) && (isNextKey)){
```
Comment (galilasmb) [inline src/main/java/br/unb/cic/analysis/df/OverridingAssignmentFieldsRefAnalysis.java:197]: Bug resolving with instance name contain $stack

File: src/test/java/br/unb/cic/analysis/samples/OverridingAssignmentObjectTwoFieldsTwoConflictsSample.java [hunk 2]
```diff
@@ -5,12 +5,12 @@
     public OverridingAssignmentInstance b;
 
     public static void main(String[] args) {
-        OverridingAssignmentObjectTwoFieldsTwoConflictsSample instanceLocal = new OverridingAssignmentObjectTwoFieldsTwoConflictsSample();
+        OverridingAssignmentObjectTwoFieldsTwoConflictsSample $stack = new OverridingAssignmentObjectTwoFieldsTwoConflictsSample();
```
Comment (galilasmb) [inline src/test/java/br/unb/cic/analysis/samples/OverridingAssignmentObjectTwoFieldsTwoConflictsSample.java:8]: Test with instance name $stack

File: src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java [hunk 3]
```diff
@@ -51,21 +49,13 @@ public OverridingAssignmentAnalysis(Body methodBody, AbstractMergeConflictDefini
 
                     for (Local useBox: getUseVariables(u)){
                         String useBoxName = getLocalName(useBox);
-                        if (defInName.equals(useBoxName) && defInName.equals(localName)){
+                        if (!defInName.equals(useBoxName) && defInName.equals(localName)){
```
Comment (barbosamaatheus) [inline src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java:52]: Because this "!" was added here?

Comment (galilasmb) [reply src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java:52]: The case x = x + ...; does not generate conflict. So the right side doesn't have to be the equals.

Comment (galilasmb) [reply src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java]: I updated this

Comment (galilasmb) [reply src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java]: There is no need to analyze the UseBox of a definition (DefBox). Example: x = ...; no matter what comes on the right of the statement.

File: src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java [hunk 4]
```diff
@@ -46,13 +46,10 @@ public OverridingAssignmentAnalysis(Body methodBody, AbstractMergeConflictDefini
 
                 for (DataFlowAbstraction defsIn: in){
                     String defInName = getLocalName(defsIn.getLocal());
-
-                    for (Local useBox: getUseVariables(u)){
-                        String useBoxName = getLocalName(useBox);
-                        if (!defInName.equals(useBoxName) && defInName.equals(localName)){
```
Comment (galilasmb) [inline src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java]: @barbosamaatheus There is no need to analyze the UseBox of a definition (DefBox). Example: x = ...; no matter what comes on the right of the statement.

File: src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java [hunk 5]
```diff
@@ -46,13 +46,10 @@ public OverridingAssignmentAnalysis(Body methodBody, AbstractMergeConflictDefini
 
                 for (DataFlowAbstraction defsIn: in){
                     String defInName = getLocalName(defsIn.getLocal());
-
-                    for (Local useBox: getUseVariables(u)){
```
Comment (galilasmb) [inline src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java:52]: UseBox of the statement, no matter...

File: src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java [hunk 6]
```diff
@@ -46,13 +46,10 @@ public OverridingAssignmentAnalysis(Body methodBody, AbstractMergeConflictDefini
...
-                        String useBoxName = getLocalName(useBox);
-                        if (!defInName.equals(useBoxName) && defInName.equals(localName)){
-                            res.add(new DataFlowAbstraction(defsIn.getLocal(), defsIn.getStmt())); //update an element in IN
-                            break; //Do not necessary check others elements
-                        }
+                    //if u not in IN, then add it
+                    if (!defInName.equals(localName)){
```
Comment (galilasmb) [inline src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java:50]: Just add, if this definition is not yet present in the IN.

**Pull request merged**